WiseUnpacker/SabreTools.Serialization.Wrappers/WiseSectionHeader.cs

```diff
@@ -214,49 +218,197 @@ private bool ExtractHeaderDefinedFiles(Stream data, string outputDirectory, bool
 
             // Does output size include the crc32? Doesn't seem to?
 
-            // Extract main MSI file
-            var expected = new DeflateInfo
-            {
-                InputSize = MsiFileEntryLength,
-                OutputSize = MsiFileEntryLength - 4,
-                Crc32 = 0
-            };
-            data.Seek(data.Position - MsiFileEntryLength, 0);
-            if (InflateWrapper.ExtractFile(data, "ExtractedMsi.msi", outputDirectory, expected, false, includeDebug)
-                == ExtractionStatus.FAIL)
+            // Comparing against extractionstatus GOOD for now.
+            
+            // Extract first executable, if it exists
+            if (ExtractFile(data, "FirstExecutable.exe", outputDirectory, FirstExecutableFileEntryLength, includeDebug)
+                != ExtractionStatus.GOOD)
                 return false;
-            data.Seek(data.Position - MsiFileEntryLength, 0);
 
             // Extract second executable, if it exists
-            expected = new DeflateInfo
-            {
-                InputSize = SecondExecutableFileEntryLength,
-                OutputSize =
-                    SecondExecutableFileEntryLength - 4,
-                Crc32 = 0
-            };
-            data.Seek(data.Position - SecondExecutableFileEntryLength, 0);
-            if (InflateWrapper.ExtractFile(data, "WiseScript.bin", outputDirectory, expected, false, includeDebug)
-                == ExtractionStatus.FAIL)
+            if (ExtractFile(data, "SecondExecutable.exe", outputDirectory, SecondExecutableFileEntryLength, 
+                includeDebug)
+                != ExtractionStatus.GOOD)
+                return false;
+            
+            // Extract second executable, if it exists
+            // If there's a size provided for the second executable but no size for the first executable, the size of
+            // the second executable appears to be some unrelated value that's larger than the second executable
+            // actually is. Currently unable to extract properly in these cases, as no header value in such installers
+            // seems to actually correspond to the real size of the second executable.
```

Comment:
No change needed, but this is oddly similar to some of the "phantom" values in the normal overlay header. I haven't figured it out there either. I just ignore it.